Code Improvements

bdlukaa opened the pull request:
Several code improvements all around the project. This implements the following things:

- Accessibility support for all the widgets
- Diagnostics support for almost all the widgets and classes (devtools)
- `debugCheckHasMacosTheme` now has a better error message
- Improved documentation all around the package
- Replaced `todo` -> `TODO`, because of dart analyzer
- Replaced `CupertinoDynamicColor` -> `DynamicColorX`
- Added `MacosTheme.maybeOf`, which can result in a nullable `MacosThemeData`

## Pre-launch Checklist

- [x] I have ran `dartfmt` on all changed files
- [x] I have ran "optimize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes

bdlukaa committed:
Documentation about checkbox

bdlukaa committed:
Documentation about radio button

bdlukaa committed:
Documentation about push button

bdlukaa committed:
Documentation about toggle switch

bdlukaa committed:
Documentation about indicators

bdlukaa committed:
Documentation about help buttons

bdlukaa committed:
Code improvements

bdlukaa committed:
Merge branch 'dev' into improvements

GroovinChip commented on lib/macos_ui.dart:
```diff
@@ -14,7 +12,6 @@ export 'package:flutter/material.dart'
         FlutterLogo,
         CircleAvatar;
 
-/// todo: package-level docs
```
Let's leave this in, because we do need package-level docs

GroovinChip commented on lib/macos_ui.dart:
```diff
@@ -1,7 +1,5 @@
 library macos_ui;
 
-export 'package:flutter/cupertino.dart'
```
This removal reminds me that we really shouldn't be exporting any non-macos_ui libraries in this file, it messes with the generation of documentation. If we need the convenience of exporting flutter libraries in one file, we should have a separate file specifically for this that is not exported in `lib/macos_ui.dart`. Perhaps a `lib/src/library.dart`.

GroovinChip replied:
I will tinker with this now

GroovinChip replied:
Made the change, running dartdoc now to see the results

GroovinChip commented on CHANGELOG.md:
```diff
@@ -1,6 +1,10 @@
+## [0.0.9+1]
+* `CapacityIndicator` colors can now be setted on its constructor
```
"setted" isn't a word, change to something like "`CapacityIndicator` colors can now be set on its constructor"

bdlukaa replied:
Oh sorry. I didn't know that

GroovinChip replied:
No worries 😊

GroovinChip reviewed (changes requested):
These changes are great, I especially like the accessibility improvements!

bdlukaa committed:
Changes

GroovinChip reviewed (changes requested):
Looking good. I will submit my own changes on this branch shortly.

GroovinChip committed:
Introduce library.dart

Moving flutter exports from macos_ui.dart to library.dart allows documentation to be generated only for files belonging to macos_ui.

GroovinChip committed:
Fix placement of package level documentation

GroovinChip committed:
Pull latest changes and resolve conflicts

# Conflicts:
#	lib/macos_ui.dart
#	lib/src/layout/scaffold.dart

GroovinChip committed:
Address pana warning

GroovinChip commented:
Other changes made:
* Updated `dart_code_metrics` to latest version
* Added a lint rule to avoid relative lib imports
